base/strings/string.jl

```diff
@@ -83,6 +78,30 @@ function String(v::Vector{UInt8})
     return str
 end
 
+"""
+    takestring!(x) -> String
+
+Create a string from the content of `x`, emptying `x`.
+
+# Examples
+```jldoctest
+julia> v = [0x61, 0x62, 0x63];
+
+julia> s = takestring!(v)
+"abc"
+
+julia> isempty(v)
+true
+```
+"""
+takestring!(v::Vector{UInt8}) = String(v)
+
+"Create a string re-using the memory, if possible.
+Mutating the memory after calling this function is undefined behaviour."
+function unsafe_takestring(m::Memory{UInt8})
+    ccall(:jl_genericmemory_to_string, Ref{String}, (Any, Int), m, length(m))
```

Comment:
Why is this missing the:

```julia
len = length(m)
len == 0 && return ""
```

Reply:
I can't remember, possibly an oversight. Added.